### PR TITLE
add alt texts, remove alt text ignore from htmlproofer container

### LINKED ISSUE #24: Fix missing alt text
We currently ignore missing img alt text when using HTMLProofer, due to a former issue that is probably resolved by now: see https://github.com/miraisolutions/techguides/issues/13#issuecomment-810083276

Meaningful alt texts should included and the `--empty-alt-ignore` removed from the Actions WF

### PR DESCRIPTION
**Pull request by Chargothrond**
Closes #24

### CODE REVIEW

File: shiny-ci-cd.Rmd [hunk 1]
```diff
@@ -57,7 +57,7 @@ cache: packages
 
 As default, Travis CI takes care of package dependency installation and performs the typical package build & check you would run locally via e.g. `devtools::check()`. Such CI pipeline is triggered by any push event on any branch on the GitHub repo, including pull requests.
 
-![](shiny-ci-cd/img/ShinyCICD_travis1.png)
+![Travis CI pipeline](shiny-ci-cd/img/ShinyCICD_travis1.png)\
```
Comment (riccardoporreca) [inline shiny-ci-cd.Rmd:60]:
```suggestion
![Default Travis Continuous Integration pipeline for an R package](shiny-ci-cd/img/ShinyCICD_travis1.png)
```
Improved text, removed trailing `\` in view of the main review comment

File: shiny-ci-cd.Rmd [hunk 2]
```diff
@@ -133,7 +133,7 @@ deploy:
 
 As visible from the run logs, all the CI/CD pipeline steps are performed, despite only the deployment step being explicitly defined.
 
-![](shiny-ci-cd/img/ShinyCICD_travis2.png)
+![Travis CI executes all pipeline steps](shiny-ci-cd/img/ShinyCICD_travis2.png)\
```
Comment (riccardoporreca) [inline shiny-ci-cd.Rmd:136]:
```suggestion
![Travis Continuous Integration / Continuous Deployment pipeline for a packaged Shiny app](shiny-ci-cd/img/ShinyCICD_travis2.png)
```
Same as a bove

File: shiny-ci-cd.Rmd [hunk 3]
```diff
@@ -180,7 +180,7 @@ knitr::read_chunk("shiny-ci-cd/actions/ci-cd.yml", labels = "ci-cd.yml")
 
 As visible from the run logs, all the CI/CD pipeline steps are performed subsequently, and are identifiable by the `name` field.
 
-![](shiny-ci-cd/img/ShinyCICD_githubactions1.png)
+![GitHub Actions pipeline steps](shiny-ci-cd/img/ShinyCICD_githubactions1.png)\
```
Comment (riccardoporreca) [inline shiny-ci-cd.Rmd:183]: Same as above 
```suggestion
![GitHub Actions Continuous Integration / Continuous Deployment pipeline for a packaged Shiny app](shiny-ci-cd/img/ShinyCICD_githubactions1.png)
```

**Comment by Chargothrond** [comment]
@riccardoporreca fig_caption doesn't work with gitbooks, see my comment on https://github.com/miraisolutions/techguides/issues/13

We would have to use the fig.alt chunk option, which is something I haven't looked into yet.

**Comment by riccardoporreca** [comment]
> @riccardoporreca fig_caption doesn't work with gitbooks, see my comment on #13

`fig_caption` works for me as follows... can you include it in index.Rmd (with the comment) and check locally? The ultimate assessment should be based on the Actions artifact
```yml
output:
  bookdown::gitbook:
    split_by: section
    # disable alt text rendered as figure caption
    fig_caption: false
    # [...]
```

If this still does not work, chunk options would not be a solution as our figures are not produced by code chunks. We might go for the CSS trick, but I am confident / hopeful the `fig_caption` would do.

**Comment by Chargothrond** [comment]
hmm, it does seem to work now, but only after I knitted both the index and shiny ci cd .rmd's multiple times. perhaps I am also missing how to properly run the full bookdown document in one go.
@riccardoporreca can you please double check? Afterwards, it should be all ready to merge.

**Comment by riccardoporreca** [comment]
@Chargothrond, we can catch-up separately on the way to render the book locally, it should be best done either at the command line via (same as in the actions workflow)

```r
bookdown::render_book("index.Rmd", "bookdown::gitbook")
```

or in RStudio with Ctrl+Shift+B.

Knitting individual files, especially chapters, might do a partial rendering, and perhaps not pick the updated configuration.

You may want to also get familiar with looking at the actions workflow artifact containing the book rendered as it will be eventually published, see e.g. https://github.com/miraisolutions/techguides/actions/runs/707817061#artifacts

I am about to merge this, will just include the comment about `fig_caption`.

File: index.Rmd [hunk 4]
```diff
@@ -7,6 +7,7 @@ site: bookdown::bookdown_site
 output:
   bookdown::gitbook:
     split_by: section
+    fig_caption: false
```
Comment (riccardoporreca) [inline index.Rmd:10]: Include comment
```suggestion
    # disable alt text rendered as figure caption
    fig_caption: false
```

**Review by riccardoporreca — APPROVED** [review]
LGTM

**Pull request merged**